### PR TITLE
avoid O(N) and fix defer reader close

### PR DESCRIPTION
**Pull request by runcom**
- First commit deals with an O(N) routine and removes it (the last layer ID index takes N-1 cycle to be found)
- Second commit remove a reader close call because the reader is used after the function returns

### CODE REVIEW

File: lib/internal/backend/repository/repository2.go [hunk 1]
```diff
@@ -512,7 +504,6 @@ func (rb *RepositoryBackend) getLayerV2(layerID string, dockerURL *types.ParsedD
 			if err != nil {
 				return nil, nil, err
 			}
-			defer res.Body.Close()
```
Comment (dgonyeo) [inline lib/internal/backend/repository/repository2.go]: I just noticed that the [original res](https://github.com/appc/docker2aci/pull/180/files#diff-acab445bd3e69965398509002dd14a18L499) doesn't get closed if this if block gets dropped into. Would you mind adding a `res.Body.Close()` around like 507 while you're here?

**Comment by dgonyeo** [comment]
LGTM

File: lib/internal/backend/repository/repository2.go (hunk 1 above)
Comment (runcom) [inline lib/internal/backend/repository/repository2.go]: there's a problem however, response body closing is an issue in all of this routine, I'll add a defer

**Comment by runcom** [comment]
@dgonyeo reworked the second commit so that if there's an error and the `res` var isn't nil then we close the body

File: lib/internal/backend/repository/repository2.go [hunk 2]
```diff
@@ -496,23 +492,32 @@ func (rb *RepositoryBackend) getLayerV2(layerID string, dockerURL *types.ParsedD
...
 	}
 
-	res, err := rb.makeRequest(req, dockerURL.ImageName, accepting)
+	res, err = rb.makeRequest(req, dockerURL.ImageName, accepting)
 	if err != nil {
 		return nil, nil, err
 	}
 
+	defer func() {
+		if err != nil {
```
Comment (iaguis) [inline lib/internal/backend/repository/repository2.go]: Just "and" those to save indentation?

**Comment by iaguis** [comment]
One nit but LGTM afterwards.

File: lib/internal/backend/repository/repository2.go (hunk 2 above)
Comment (runcom) [inline lib/internal/backend/repository/repository2.go]: changed to and

**Comment by iaguis** [comment]
Thanks!

**Pull request merged**